parse_example.py

```diff
@@ -0,0 +1,70 @@
+import csv
+import numpy
+import json
+from json import dumps
+import geocoder
+
+
+MYFILE = "....repro.csv"
+
+#Stage 1: Parse your data
+class Parse (object):
+    def __init__ (self, parsed_file = None):
+        self.parsed_file = parsed_file
+
+    def parse_data (self, raw_file, delimiter = ','):
+        """Parse our CSV file and make it into data!"""
+        #open our file
+        opened_file = open (raw_file)
+        #read our file
+        csv_reader = csv.reader (opened_file, delimiter = delimiter)
+        #convert contents of our file to a dict stored in an list
+        parsed_data = []
+        headers = csv_reader.next ()
+        for items in csv_reader:
+            parsed_data.append (dict(zip(headers,items)))
+        #close file
+        opened_file.close ()
+        return parsed_data
+
+    def saveparse (self, raw_file):
+        """Parses your raw file and writes it to a new CSV file."""
+        with open ('parsed_file.csv', 'w') as new_file:
+            #write contents of parsed file to new file and close
+            parsed_data = self.parse_data(raw_file)
+            new_file.write (str(parsed_data))
+
+def create_map (data_file):
+    """Take our parsed data and geocode it to a map."""
+    geo_map = {'type': 'FeatureCollection'}
+    list_of_items = []
+    for index, item in enumerate(data_file):

```

Comment:
Advanced stuff.

Consider using list comprehensions for this: http://www.secnetix.de/olli/Python/list_comprehensions.hawk

Okay, more detail.

So the typical way, when learning to code, is to build an iterative algorithm. Something like this.

1. Initialize a new empty list for results.
2. Put all my data into a different list of stuff.
3. Walk over the list of stuff. For each element in the list:
  a. Do some stuff to the data and put it into a temporary variable
  b. Put the value in the temporary variable into the results list.
4. Return the results list.

This is all well and good, and is a good place to start. But another way that you can think about this is the idea of "list comprehensions" or "map" -- where "map" is the same as in Google's "map-reduce" algorithm.

Essentially what you do is, you take your list and say: "Apply this function to every element of this list, and return a new list." It's a "transform", in other words -- you take some list of things, and turn it into another list of equal length, but each element in that list is "transformed."

Being able to think this way allows you to conceptualize the flow of data in a way that makes a bit more sense -- the idea being that your data passes through a series of transforms and filters until you reach the endpoint that you want your data to look like.

In this case, you'd have an algorithm that looks something like this.

1. Take all the data in the data file.
2. Remove everything that has no valid Country. (Think of this as a "filter". The `pydash` library http://pydash.readthedocs.io/en/latest/api.html#pydash.collections.filter_ can help with this.)
3. Apply a transform function via list comprehension to the intermediate data to turn each entry into what it should look like.
4. Return the transformed list.